packages/pwa-kit-dev/src/configs/webpack/loaders/extension-loader.js

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2024, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {getConfig} from '@salesforce/pwa-kit-runtime/utils/ssr-config'
+import dedent from 'dedent'
+import {kebabToUpperCamelCase} from '../utils'
+
+const APP_EXTENSION_CLIENT_ENTRY = 'setup-app'
+
+module.exports = function () {
+    // NOTE: We need to account for extensions being tuples. Here we assume it's a simple
+    // string that is the npm package name.
+    const {extensions = []} = getConfig()?.app || {}
+
+    return dedent`
+            /*
+            * Copyright (c) 2024, salesforce.com, inc.
+            * All rights reserved.
+            * SPDX-License-Identifier: BSD-3-Clause
+            * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+            */
+            // Extension Imports
+            ${extensions
+                .map(
+                    (extension) =>
+                        `import ${kebabToUpperCamelCase(
+                            extension.split('/')[1]
+                        )} from '${extension}/${APP_EXTENSION_CLIENT_ENTRY}'`
+                )
+                .join('\n            ')}
+            
+            export default [
+                ${extensions
+                    .map((extension) => {
+                        const config = {} // TODO: Parse config config here.
+                        return `new ${kebabToUpperCamelCase(
+                            extension.split('/')[1]
+                        )}(${JSON.stringify(config)})`
+                    })
+                    .join(',\n                ')}
+            ]
+        `
```

Comment:
it would be nice to document the "why".

I think this is because we need the import statements for webpack to correctly build stuff, but I think we may want to document all the details, it could be very useful when we look at this in the future.

Reply:
Good point. This is kinda the linch pin of the solution, so it makes sense to document it a little more robustly.

Reply:
I added a blurb.